src/utils.jl

```diff
@@ -108,16 +111,19 @@ function diff2term(O)
     else
         ds = nothing
     end
-    T = symtype(O)
+
     if ds === nothing
         return similarterm(O, operation(O), map(diff2term, arguments(O)), metadata=metadata(O))
     else
         oldop = operation(O)
-        if !(oldop isa Sym)
+        if oldop isa Sym
+            opname = string(nameof(oldop))
+        elseif oldop isa Term && operation(oldop) === getindex
```

Comment:
it maybe better to use `issym` and `isterm` in these if statement because we're going to be soon not using these types...